packages/components/src/form/index.js

```diff
@@ -33,18 +32,28 @@ class Form extends Component {
 		return ! Object.keys( this.state.errors ).length;
 	}
 
-	validate() {
+	validate( onValidate = () => {} ) {
 		const { values } = this.state;
 		const errors = this.props.validate( values );
-		this.setState( { errors } );
+		this.setState( { errors }, onValidate );
 	}
 
 	setValue( name, value ) {
 		this.setState(
 			( prevState ) => ( {
 				values: { ...prevState.values, [ name ]: value },
 			} ),
-			this.validate
+			() => {
+				this.validate( () => {
+					// onChangeCallback keeps track of validity, so needs to
+					// happen after setting the error state.
+					this.props.onChangeCallback(
```

Comment:
This is a handy new callback on form that lets you react to individual value changes and check validity per value change rather than just on submit.